pkgs/development/python-modules/torch/default.nix

```diff
@@ -80,24 +73,104 @@
 
   # ROCm dependencies
   rocmSupport ? config.rocmSupport,
-  rocmPackages_5,
+  rocmPackages,
   gpuTargets ? [ ],
 }:
 
 let
-  inherit (lib)
-    attrsets
-    lists
-    strings
-    trivial
+  inherit (lib.attrsets) optionalAttrs;
+  inherit (lib.lists)
+    concatMap
+    intersectLists
+    map
+    optionals
+    subtractLists
+    ;
+  inherit (lib.meta) getExe';
+  inherit (lib.strings)
+    concatStringsSep
+    optionalString
+    replaceChars
+    versionAtLeast
+    versionOlder
+    ;
+  inherit (lib.trivial) id throwIf;
+  inherit (lib.versions) major majorMinor;
+
+  inherit (stdenv)
+    cc
+    isDarwin
+    isLinux
+    isx86_64
+    ;
+  inherit (stdenv.hostPlatform) darwinSdkVersion;
+
+  inherit (cudaPackages)
+    cuda_cccl
+    cuda_cudart
+    cuda_cupti
+    cuda_nvcc
+    cuda_nvml_dev
+    cuda_nvprof
+    cuda_nvrtc
+    cuda_nvtx
+    cuda_profiler_api
+    cudaAtLeast
+    cudaFlags
+    cudaMajorMinorVersion
+    cudaOlder
+    libcublas
+    libcufft
+    libcurand
+    libcusolver
+    libcusparse
+    nccl
```

Comment:
Honestly, I think in this case `with;` and `cudaPackages.${name}` are more reasonable